script_num.c

```diff
@@ -34,19 +34,14 @@ sn_from_int(s64 n, struct num *sn) {
 void sn_serialize(struct num *sn, u8 *buf, int bufsize, u16 *len) {
   u8 *p = buf;
 
-  if(sn->val == 0) {
-    *len = 0;
-    return;
-  }
-
   const int neg = sn->val < 0;
   u64 absvalue = neg ? -(sn->val) : sn->val;
 
-  while(absvalue) {
+  do {
     *(p++) = absvalue & 0xff;
     assert((p - buf) <= bufsize);
     absvalue >>= 8;
-  }
+  } while(absvalue);
```

Comment:
why was this changed, it looks wrong?

Reply:
For the error case: `stack_hex                                {missing 00}` where 0 values are not displayed.  I changed the while loop to a do loop.  Further up in that function, when you have absvalue=0, it skips over the while loop and therefore the 0 is not serialized and not displayed.

Reply:
It fixes these failing tests:

    not ok 17 - test_OP_NOT_1
    #
    #    got      'script      1 OP_NOT script_hex  5191 stack       0 stack_hex    results     success', 
    #    expected 'script      1 OP_NOT script_hex  5191 stack       0 stack_hex   00 results     success'
    #
    not ok 18 - test_OP_NOT_2
    #
    #    got      'script      2 OP_NOT script_hex  5291 stack       0 stack_hex    results     success', 
    #    expected 'script      2 OP_NOT script_hex  5291 stack       0 stack_hex   00 results     success'
    #
    not ok 19 - test_OP_0NOTEQUAL0
    #
    #    got      'script      0 OP_0NOTEQUAL script_hex  0092 stack       0 stack_hex    results     success', 
    #    expected 'script      0 OP_0NOTEQUAL script_hex  0092 stack       0 stack_hex   00 results     success'
    #
    not ok 23 - test_BOOLAND
    #
    #    got      'script      0 0 OP_BOOLAND script_hex  00009a stack       0 stack_hex    results     success', 
    #    expected 'script      0 0 OP_BOOLAND script_hex  00009a stack       0 stack_hex   00 results     success'
    #

Reply:
ok I'll take a closer look soon